### ISSUE TITLE
Hi author, is there any problem with your code below? Regarding the dimension of labels, after going through labels=labels[: , 0, :], shouldn't labels have become two-dimensional? 

### ISSUE THREAD

**Issue by Pythonerzan** [happyalfred2016/3D-Mixer#1]
for i in range(labels.shape[1]):
                assert outputs[i].shape == labels[:, i, :].shape
                non_nan = ~torch.isnan(labels[:, i, :])
                if non_nan.any():
                    loss = self.criterion(outputs[i][non_nan], labels[:, i, :][non_nan])
                    loss.backward(retain_graph=True)
                    losses[i] += loss.detach()

**Comment by happyalfred2016** [comment]
Hi, sorry for the late response.

It's because, before "labels=labels[:, 0, :]", the shape of labels is [b, p, 1, 1], where b and p denotes batch size and the number of patches. And actually "labels=labels[:, 0, :]" equals to "labels=labels[:, 0, :, :]"

**Comment by Pythonerzan** [comment]
> Hi, sorry for the late response.
> 
> It's because, before "labels=labels[:, 0, :]", the shape of labels is [b, p, 1, 1], where b and p denotes batch size and the number of patches. And actually "labels=labels[:, 0, :]" equals to "labels=labels[:, 0, :, :]"

Thank you so much for your prompt and helpful response! I appreciate your time and effort in explaining the issue to me.

**Comment by happyalfred2016** [comment]
😊
